nixos/modules/services/web-apps/filebrowser.nix

```diff
@@ -0,0 +1,181 @@
+      key = cfg.tlsCertificateKey;
+    })
+    // (optionalAttrs (cfg.baseUrl != null) { baseurl = cfg.baseUrl; })
+    // (optionalAttrs (cfg.cacheDir != null) { cache-dir = cfg.cacheDir; })
+    // (optionalAttrs cfg.disableThumbnails { disable-thumbnails = true; })
+    // (optionalAttrs cfg.disablePreviewResize { disable-preview-resize = true; })
+    // (optionalAttrs cfg.disableCommandRunner { disable-exec = true; })
+    // (optionalAttrs cfg.disableTypeDetectionByHeader { disable-type-detection-by-header = true; })
+  ));
+in
+{
+  options.services.filebrowser = {
+    enable = mkEnableOption
+      "File Browser is a web application for managing files and directories";
+
+    package = mkPackageOption pkgs "filebrowser" { };
+
+    # https://github.com/filebrowser/filebrowser/blob/bd3c1941ff8289a5dae877e08f7e25fa9b2a92c5/cmd/root.go#L56
+    address = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      example = literalExpression "0.0.0.0";
+      description = ''
+        Address the service should listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      description = "Port the service should listen on.";
+    };
+
+    tlsCertificate = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "Optional TLS certificate";
+    };
+
+    tlsCertificateKey = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Key for TLS certificate. Must be set if `services.filebrowser.tlsCertficate`
+        is used.
+      '';
+    };
+
+    rootDir = mkOption {
+      type = types.path;
+      default = "/var/lib/filebrowser/files";
+      description = ''
+        Path to the directory that should be exposed by File Browser.
+      '';
+    };
+
+    baseUrl = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "/files";
+      description = ''
+        Subpath to serve the web interface at. Useful in combination with a
+        reverse proxy.
+      '';
+    };
+
+    cacheDir = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      example = literalExpression "/var/cache/filebrowser";
+      description = "File cache directory. Disabled by default.";
+    };
+
+    tokenExpirationTime = mkOption {
+      type = types.str;
+      default = "2h";
+      example = "12h";
+      description = "User session timeout";
+    };
+
+    imageProcessors = mkOption {
+      type = types.int;
+      default = 4;
+      example = 2;
+      description = "Number of image processes.";
+    };
+
+    disableThumbnails = mkEnableOption "Disable image thumbnails.";
+    disablePreviewResize = mkEnableOption "Disable resize of image previews.";
+    disableCommandRunner = mkEnableOption "Disable the Command Runner feature.";
+    disableTypeDetectionByHeader = mkEnableOption
+      "Disable file type detection by reading file headers.";
```

Comment:
The description of those will be prefixed with ``Wether to enable`` which I don't think the description considered.